src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPi3Driver.Linux.cs

```diff
@@ -358,13 +405,60 @@ private void Initialize()
                     return;
                 }
 
-                int fileDescriptor = Interop.open(GpioMemoryFilePath, FileOpenFlags.O_RDWR | FileOpenFlags.O_SYNC);
+                // try and open /dev/gpiomem
+                fileDescriptor = Interop.open(GpioMemoryFilePath, FileOpenFlags.O_RDWR | FileOpenFlags.O_SYNC);
                 if (fileDescriptor == -1)
                 {
-                    throw new IOException($"Error {Marshal.GetLastWin32Error()} initializing the Gpio driver.");
+                    win32Error = Marshal.GetLastWin32Error();
+
+                    // if the failure is NOT because /dev/gpiomem doesn't exist then throw an exception at this point.
+                    // if it were anything else then it is probably best not to try and use /dev/mem on the basis that
+                    // it would be better to solve the issue rather than use a method that requires root privileges
+                    if (win32Error != Linux_Enoent)
+                    {
+                        throw new IOException($"Error {win32Error} initializing the Gpio driver.");
+                    }
+
+                    // if /dev/gpiomem doesn't seem to be available then let's try /dev/mem
+                    fileDescriptor = Interop.open(MemoryFilePath, FileOpenFlags.O_RDWR | FileOpenFlags.O_SYNC);
+                    if (fileDescriptor == -1)
+                    {
+                        throw new IOException($"Error {Marshal.GetLastWin32Error()} initializing the Gpio driver.");
+                    }
+                    else // success so set the offset into memory of the gpio registers
+                    {
+                        gpioRegisterOffset = ~0U;
```

Comment:
use explicit value

Reply:
Done in the PR Update

Reply:
still here (I think you only updated the one below)

Reply:
I think I have them all in the next PR Update